test/e2e_node/pod_hostnamefqdn_test.go

```diff
@@ -140,4 +148,53 @@ var _ = SIGDescribe("Hostname of Pod [Feature:SetHostnameAsFQDN][NodeAlphaFeatur
 		f.TestContainerOutput("fqdn and fqdn", pod, 0, output)
 	})
 
+	/*
+	   Release: v1.19
+	   Testname: Fail to Create Pod with longer than 64 bytes FQDN when setHostnameAsFQDN field set to true.
+	   Description: A Pod that defines the subdomain field in it spec has FQDN.
+	                 When setHostnameAsFQDN: true, the hostname is set to be
+	                 the FQDN. Since kernel limit is 64 bytes for hostname field,
+	                 if pod FQDN is longer than 64 bytes it will generate events
+	                 regarding FailedCreatePodSandBox.
+	*/
+
+	ginkgo.It("a pod configured to set FQDN as hostname will remain in Pending "+
+		"state generating FailedCreatePodSandBox events when the FQDN is "+
+		"longer than 64 bytes", func() {
+		// 55 characters for name plus -<int>.t.svc.cluster.local is way more than 64 bytes
+		pod := testPod("hostfqdnveryveryveryverylongforfqdntobemorethan64bytes")
+		pod.Spec.Containers[0].Command = []string{"sh", "-c", "echo $(hostname)';'$(hostname -f)';'"}
+		subdomain := "t"
+		// Set PodSpec subdomain field to generate FQDN for pod
+		pod.Spec.Subdomain = subdomain
+		// Set PodSpec setHostnameAsFQDN to set FQDN as hostname
+		setHostnameAsFQDN := true
+		pod.Spec.SetHostnameAsFQDN = &setHostnameAsFQDN
+		// Create Pod
+		launchedPod := f.PodClient().Create(pod)
+		// Pod should remain in the pending state generating events with reason FailedCreatePodSandBox
+		// Expected Pod FQDN
+		hostFQDN := fmt.Sprintf("%s.%s.%s.svc.%s", launchedPod.ObjectMeta.Name, subdomain, launchedPod.ObjectMeta.Namespace, dnsDomain)
+		// Expected Message Error Event
+		expectedMessage := fmt.Sprintf("Failed to create pod sandbox: failed "+
```

Comment:
for my taste it's a little bit too specific of a message. I guess it's fine since this test run as part of PR verification.

Reply:
should I just match the beginning without including variables? For example: "Failed to create pod sandbox: failed to construct FQDN from pod hostname and cluster domain"

Reply:
I'd like this more. Maybe even just the word "fqdn". But again, this is the matter of taste. Important things test validates is that pod is in pending state and reported event. Specific text shouldn't matter that much.

Reply:
ok, I updated to just look for the first part of the string.